source/distributed/tria.cc

```diff
@@ -1124,10 +1124,545 @@ namespace
+                cumulative_sizes_fix.back() - cumulative_sizes_fix.front();
+
+              src_data_fix.insert(src_data_fix.end(),
+                                  bytes_skipped,
+                                  static_cast<char>(-1)); // invalid_char
+            }
+        }
+
+      // Double check that we packed everything correctly.
+      Assert(src_data_fix.size() == src_data_fix.capacity(),
+             ExcInternalError());
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::DataTransfer::execute_transfer(
+      const typename dealii::internal::p4est::types<dim>::forest
+        *parallel_forest,
+      const typename dealii::internal::p4est::types<dim>::gloidx
+        *previous_global_first_quadrant)
+    {
+      Assert(cumulative_sizes_fix.size() > 0,
+             ExcMessage("No data has been packed!"));
+
+      // Resize memory according to the data that we will receive.
+      dest_data_fix.resize(parallel_forest->local_num_quadrants *
+                           cumulative_sizes_fix.back());
+
+      // Execute fixed size transfer.
+      dealii::internal::p4est::functions<dim>::transfer_fixed(
+        parallel_forest->global_first_quadrant,
+        previous_global_first_quadrant,
+        parallel_forest->mpicomm,
+        0,
+        dest_data_fix.data(),
+        src_data_fix.data(),
+        cumulative_sizes_fix.back());
+
+      // Release memory of previously packed data.
+      src_data_fix.clear();
+      src_data_fix.shrink_to_fit();
+
+      // TODO: for variable transfer
+      // allocate memory for transfer
+      // p4est transfer fixed_begin for transfer of var data sizes
+      // p4est transfer custom for transfer of var data
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::DataTransfer::unpack_cellstatus(
+      std::vector<quadrant_cell_relation_t> &quad_cell_relations) const
+    {
+      Assert(cumulative_sizes_fix.size() > 0,
+             ExcMessage("No data has been packed!"));
+      Assert((quad_cell_relations.size() == 0) || (dest_data_fix.size() > 0),
+             ExcMessage("No data has been received!"));
+
+      // Size of CellStatus object that will be unpacked on each cell.
+      const size_t size = sizeof(
+        typename parallel::distributed::Triangulation<dim,
+                                                      spacedim>::CellStatus);
+
+      // Iterate over all cells and overwrite the CellStatus
+      // information from the transferred data.
+      // Proceed buffer iterator position to next cell after
+      // each iteration.
+      auto it_qcrel     = quad_cell_relations.begin();
+      auto cit_dest_fix = dest_data_fix.cbegin();
+      for (; it_qcrel != quad_cell_relations.end();
+           ++it_qcrel, cit_dest_fix += cumulative_sizes_fix.back())
+        {
+          auto &cell_status = std::get<1>(*it_qcrel);
+
+          cell_status =
```

Comment:
this is a bit difficult to read -- it looks like you're only ever writing into a local variable, with no external change, if one misses the `&` in the declaration of the variable. Just put the `std::get<1>(*it_qcrel)` on the lhs here and drop the declaration of the `cell_status` variable. (but maybe say that the result of `get<1>` is the cell status.)

Reply:
actually, you're already saying this in the comment above, so never mind about the last part of my comment